Engine/Component/ComponentTrail.cpp

```diff
@@ -264,13 +269,17 @@ void ComponentTrail::Render()
 		glBindBuffer(GL_UNIFORM_BUFFER, 0);
 
 		glUniform4fv(glGetUniformLocation(shader_program, "color"), 1, color.ptr());
-
+		glUniform4fv(glGetUniformLocation(shader_program, "color_blend"), 1, color_to_blend.ptr());
 		glUniform1f(glGetUniformLocation(shader_program, "emissive_intensity"), emissive_intensity);
+		glUniform1f(glGetUniformLocation(shader_program, "percentage"), blend_percentage);
+		glUniform1f(glGetUniformLocation(shader_program, "smooth_step"), smoothening_step);
 
 		glDrawArrays(GL_TRIANGLE_STRIP, 0, vertices.size());
 		glBindVertexArray(0);
 
 		glUseProgram(0);
+		glDisable(GL_BLEND);
```

Comment:
Becareful with this, as your disabling Blending without knowing if its needed in next outside renderign operations. My advice here will be getting the current blending status (look at `glIsEnabled`), doing whatever you want in the render part and at the end turning the `GL_BLEND` status as it was previously